src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Utils/ElementSerializer.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Xml.Linq;
+using Microsoft.Sbom.Parsers.Spdx30SbomParser.Entities;
+
+public class ElementSerializer : JsonConverter<List<Element>>
+{
+    public override List<Element> Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        throw new NotImplementedException("Deserialization of Elements into specific subtypes is not implemented yet.");
+    }
+
+    public override void Write(Utf8JsonWriter writer, List<Element> elements, JsonSerializerOptions options)
+    {
+        writer.WriteStartArray();
+
+        foreach (var element in elements)
+        {
+            switch (element.Type)
+            {
+                case "AnyLicenseInfo":
+                    JsonSerializer.Serialize(writer, (AnyLicenseInfo)element, options);
```

Comment:
Serialize has a templated version that might be slightly cleaner--note that this pattern would apply to all of these:
```suggestion
                    JsonSerializer.Serialize<AnyLicenseInfo>(writer, element at AnyLicenseInfo, options);
```

Part of me wants to see us use a visitor pattern here instead of a switch statement, but that may not be worth the cost

Reply:
This is no longer relevant with the new changes.